### DISCUSSION TITLE
Better paste indentation support

### DISCUSSION THREAD

**Discussion by godismyjudge95** [nvim-mini/mini.nvim#447] [Ideas]
Perhaps a good misc addition or a separate module altogether?

https://github.com/ku1ik/vim-pasta

**Comment by echasnovski** [comment]
Looks interesting, thanks for suggestion!

On one hand, looking at implementation it might indeed be a good fit for 'mini.misc'. On the other, I personally need more convincing to override such basic mappings as `p` and `P` and making them exported functions is not that user friendly.

Also, you can achieve a very similar functionality by tweaking `o`/`O` instead and use `grr` from [mini.operators](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-operators.md):

```vim
nnoremap <Leader>o o<Space><BS><Esc>
nnoremap <Leader>O O<Space><BS><Esc>
```

These variants will create new line below/above current line with "correct" indentation and ending up in Normal mode. After that, doing `grr` will replace this line with proper register and `[count]`.

**Comment by godismyjudge95** [comment]
Hmmm, I think this would work for my usecase but I can't seem to get it to work:
```
vim.keymap.set('n', '<leader>o', 'o<Space><BS><Esc>grr', {
    desc = 'Paste after with proper indentation',
    noremap = true,
    silent = true
})
```

This creates a new line with proper indentation but then just types "r".  If I do the keystrokes myself it works - pasting in the content with the proper indentation.  Any ideas?

**Reply by echasnovski** [reply]
> Any ideas?

Instead of `noremap = true` use `remap = true`, because you'd actually want to use meaning of the keys as they are mapped and not as they are coming from default mappings.

**Reply by godismyjudge95** [reply]
That does the trick thanks!  And the indentation works fine.
